File: app/src/main/java/com/zhiyong/tingxie/ui/answer/AnswerViewModel.java
package com.zhiyong.tingxie.ui.answer;

import android.app.Application;

import androidx.lifecycle.AndroidViewModel;
import androidx.annotation.NonNull;

import com.zhiyong.tingxie.QuizRepository;
import com.zhiyong.tingxie.db.Question;
import com.zhiyong.tingxie.db.Quiz;
import com.zhiyong.tingxie.ui.main.QuizItem;

public class AnswerViewModel extends AndroidViewModel {
    private QuizRepository mRepository;

    public AnswerViewModel(@NonNull Application application) {
        super(application);
        mRepository = new QuizRepository(application, -1);
    }

    void insertQuestion(Question question) {
        mRepository.insertQuestion(question);
    }

    void updateQuiz(QuizItem quizItem) {
        mRepository.updateQuiz(new Quiz(
                quizItem.getId(),
                quizItem.getDate(),
                quizItem.getTitle(),
                quizItem.getTotalWords(),
                quizItem.getNotLearned(),
                quizItem.getRound()
        ));
    }
}

File: app/src/main/java/com/zhiyong/tingxie/ui/answer/AnswerViewModel.kt
package com.zhiyong.tingxie.ui.answer

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.viewModelScope
import com.zhiyong.tingxie.QuizRepository
import com.zhiyong.tingxie.network.NetworkQuiz
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

class AnswerViewModel(application: Application) : AndroidViewModel(application) {
  private val repository = QuizRepository(application)

  fun upsertCorrectRecord(wordId: Long): LiveData<Int> {
    val result = MutableLiveData<Int>()
    viewModelScope.launch(Dispatchers.IO) {
      val numRecords = repository.upsertCorrectRecord(wordId)
      result.postValue(numRecords)
    }
    return result
  }

  fun updateQuiz(quizItem: NetworkQuiz): LiveData<Int> {
    val result = MutableLiveData<Int>()
    viewModelScope.launch(Dispatchers.IO) {
      val numRecords = repository.updateQuiz(quizItem)
      result.postValue(numRecords)
    }
    return result
  }
}
